cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/preview/DefaultPreviewManager.java

```diff
@@ -215,6 +219,11 @@ public ApplicationId start(@Name("namespaceId") NamespaceId namespace, AppReques
       throw new IllegalStateException("Preview service is not running. Cannot start preview for " + programId);
     }
 
+    LinkedTreeMap<String, Double> config = (LinkedTreeMap<String, Double>) previewRequest.getAppRequest().getConfig();
```

Comment:
Any reason why you are using the `LikedTreeMap` implementation class instead of the `Map` interface?

Reply:
Also, this casting is unsafe. Why need casting?

Reply:
@fernst I had initially used the type that I saw in the debugger, updated to a map now. @chtyim The getConfig() method returns a generic type hence cannot compile the code without casting it.